server/src/signin-button.js

```diff
@@ -2,44 +2,64 @@ const React = require("react");
 const PropTypes = require("prop-types");
 const { Localized } = require("fluent-react/compat");
 const sendEvent = require("./browser-send-event.js");
+const { FxaOnboardingDialog } = require("./fxa-onboarding-dialog");
 
 exports.SignInButton = class SignInButton extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       displaySettings: props.isFxaAuthenticated,
+      showFxaOnboardingDialog: !!props.showFxaOnboardingDialog,
     };
   }
 
   static getDerivedStateFromProps(nextProps, prevState) {
-    return { displaySettings: nextProps.isFxaAuthenticated };
+    return { displaySettings: nextProps.isFxaAuthenticated,
+             showFxaOnboardingDialog: nextProps.showFxaOnboardingDialog};
```

Comment:
This is needed to load correct state for myshots where firstRun doesn't load the onboardingDialog state, however this is interfering with click of dismiss logic for home and shot page